sdks/besu/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -0,0 +1,164 @@
+    lockInfoHTLC.setHashbase64(Buffer.from(hash.getSerializedPreimageBase64()));
+    lockInfoHTLC.setExpirytimesecs(expiryTimeSecs);
+    lockInfoHTLC.setTimespec(assetLocksPb.TimeSpec.EPOCH)
+    return Buffer.from(lockInfoHTLC.serializeBinary())
+}
+
+// Create an asset claim structure
+function createAssetClaimInfoSerialized(hash: Hash) {
+    const claimInfoHTLC = new assetLocksPb.AssetClaimHTLC();
+    claimInfoHTLC.setHashmechanism(hash.HASH_MECHANISM);
+    claimInfoHTLC.setHashpreimagebase64(Buffer.from(hash.getSerializedPreimageBase64()));
+    const claimInfoHTLCSerialized = claimInfoHTLC.serializeBinary();
+    const claimInfo = new assetLocksPb.AssetClaim();
+    claimInfo.setLockmechanism(assetLocksPb.LockMechanism.HTLC);
+    claimInfo.setClaiminfo(claimInfoHTLCSerialized);
+    return Buffer.from(claimInfo.serializeBinary())
+}
+
+const createHTLC = async (
+    assetManagerContract: any,
+    tokenContract: any,
+    assetID: string,
+    assetData: string,
+    assetAmount: number,
+    senderAddress: string,
+    recipientAddress: string,
+    expiryTimeSecs: number,
+    hash: Hash
+): Promise< {result: any} > => {
+
+    if (!assetManagerContract) {
+        console.log("Contract handle not supplied");
+        return {result: false};
+    }
+    if (!recipientAddress) {
+        console.log(`Recipient address not supplied ${recipientAddress}`);
+        return {result: false};
+    }
+    const currTimeSecs = Math.floor(Date.now() / 1000);   // Convert epoch milliseconds to seconds
+    if (expiryTimeSecs <= currTimeSecs) {
+        console.log("Supplied expiry time invalid or in the past: %s; current time: %s", new Date(expiryTimeSecs).toISOString(), new Date(currTimeSecs).toISOString());
+        return {result: false};
+    }    
+
+    const protobufParams = createAssetExchangeAgreementSerialized("", String(assetID), recipientAddress.slice(2), Web3.utf8ToHex(String(assetData)));
+    const lockInfoParams = createAssetLockInfoSerialized(hash, expiryTimeSecs);
+    // Normal invoke function
+    var lockStatus = await assetManagerContract.lockAsset(
+        protobufParams,
+        tokenContract.address,
+        assetAmount,
+        lockInfoParams,
+        Web3.utf8ToHex(assetData),
+        {
+            from: senderAddress,
+        }
+    ).catch(function (e:any) {
+        console.log(e);
+        console.log("lockAsset threw an error");
+        lockStatus = {result: false}
+    })
+    return {result: lockStatus};
+};
+
+/**
+ * Latter step of a Hashed Time Lock Contract
+ * - Claim a unique asset instance using a hash preimage
+ **/
+const claimAssetInHTLC = async (
+    lockContractId: string,
+    assetManagerContract: any,
+    senderAddress: string,
+    preimage: Buffer, // pass hash
+): Promise<any> => {
+
+    if (!assetManagerContract) {
+        console.log("Contract Address not supplied");
+        return false;
+    }
+    
+    const hash = new SHA256()
```

Comment:
when you take this `hash` as param, you don't need to then have this code (currently this means only SHA256 is used, but we want user to be in control of choosing hash mechanism)